blockprotocol/blockprotocol
# Reapply first WordPress migration for users of older version

## 🌟 What is the purpose of this PR?



This PR adds a new migration that simply re-runs the first (through the `IF NOT EXISTS` clause) to ensure that old users have a database. We previously didn't bail if the DB wasn't applied correctly, which means some users are stuck with unmigrated DBs at `migration_version` 2

I've also changed the migration code to be automated rather than manual and error prone.

Please do test this locally. I may have missed some edge cases. Also feel free to push commits if this doesn't get approved/merged today :)

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1204086167143889/1204298348614666/f) _(internal)_

## 🚫 Blocked by





## 🔍 What does this change?




- Add new migration code for automatic migration management
- Drive-by add migration version to the sentry errors/telemetry data
- Add `migration_2` that reapplies `migration_1`

## 📜 Does this require a change to the docs?



## ⚠️ Known issues


This isn't much of a direct issue, but we have to account for the previous version numbering in the new migration code, so we are depending on incrementing the version numbers somewhat unintuitively. This is to have it fit the former migration code.

## 🐾 Next steps



## 🛡 What tests cover this?



- Manual testing

## ❓ How to test this?



1.  Set up the WordPress instance locally on **`main`**. Install + activate the plugin. Go to the setting page of the plugin
1.  Check DB for `block_protocol_db_migration_version` in the `wp_options` table. It should be `2`
1.  Checkout the branch
1.  Reload your setting page
1.  Confirm that the `block_protocol_db_migration_version` is now `3`

## 📹 Demo
